compair/static/modules/comment/comment-form-partial.html

```diff
@@ -22,35 +23,45 @@ <h3 class="text-center"><compair-logo type="'scale'"></compair-logo> Your Answer
         <div class="form-group">
-            <label ng-if="!selfEvalComment">Text</label>
+            <label ng-if="!selfEvalComment" class="required-star">Your feedback to <span ng-if="parent.user.id != loggedInUserId">this author</span><span ng-if="parent.user.id == loggedInUserId">your own answer</span></label>
             <label ng-if="selfEvalComment" class="required-star">Your private feedback to yourself</label>
-            <textarea ng-model="comment.content" id="commentContent" class="form-control" rows="8"
-                      ckeditor="editorOptions" required>
-            </textarea>
+            <div ng-class="{'has-error': !comment.content && saveFeedbackAttempted}">
+                <textarea ng-model="comment.content" id="commentContent" class="form-control" rows="8"
+                          ckeditor="editorOptions" required>
+                </textarea>
+            </div>
         </div>
         <div class="form-group" ng-if="answerComment">
             <input id="comment_type" type="checkbox" ng-model="comment.comment_type" ng-disabled="!(comment.comment_type == AnswerCommentType.public || comment.comment_type == AnswerCommentType.private)"
                    ng-true-value="'{{AnswerCommentType.public}}'" ng-false-value="'{{AnswerCommentType.private}}'">
             <label for="comment_type"><span class="text-muted not-bold" ng-if="!(comment.comment_type == AnswerCommentType.public || comment.comment_type == AnswerCommentType.private)">(cannot show this feedback to the whole class)</span> <span ng-if="comment.comment_type == AnswerCommentType.public || comment.comment_type == AnswerCommentType.private">Show this feedback to the whole class</span></label>
-
         </div>
     </fieldset>
-    <p class="text-center text-muted"><span class="required-star "></span> = required (please make sure these areas are filled in)</p>
+
+    <!-- different helper messages for pre or post save attempts -->
+    <p class="text-center text-muted" ng-if="!saveFeedbackAttempted"><span class="required-star "></span> = required (please make sure these areas are filled in)</p>
+    <p class="text-center text-warning" ng-if="saveFeedbackAttempted"><strong><i class="glyphicon glyphicon-warning-sign"></i></strong> {{helperMsg}}</p>
+
     <div class="form-group sub-step text-center" ng-show="comment.draft || preview">
         <input type="submit" class="btn btn-default btn-lg" value="Save Draft"
-            ng-click="comment.draft = true"
+            ng-click="showErrors($event, comment.content); comment.draft = true"
             ng-disabled="submitted" />
         &nbsp;&nbsp;
         <input type="submit" class="btn btn-success btn-lg" value="Submit"
-            ng-click="comment.draft = false"
-            ng-disabled="commentForm.$invalid || submitted" />
+            ng-click="showErrors($event, comment.content); comment.draft = false"
```

Comment:
I probably missed this in previous review.  The flow seems strange.

When opening self-eval for the first time, there are two buttons: "save draft" and "submit".  Clicking either one will have the same effect of saving the self-eval as draft.  After saving the draft, only the "save" button will be shown.  Clicking that will submit the self-eval.

Reply:
Good catch, thank you. Have pushed another commit.